src/dune_pkg/opam_repo.mli

```diff
@@ -13,27 +13,39 @@ end
 
 val equal : t -> t -> bool
 
-(** [of_opam_repo_dir_path opam_repo_dir] creates a repo representedy by a local
+module Source : sig
+  type committish = Commit of string
+
+  type t =
+    { url : string
```

Comment:
What about just using `OpamUrl.t` instead of string? Sure, this type isn't ideal, but it's better than a string.

Reply:
The problem is that an `OpamUrl.t` is not really the kind of URL that we want, since it also includes hashes which are used for branches/commits so I want to have the "convert OpamURL to a thing that the Rev Store understands" in one place. With commits its easy so far, but OpamURL doesn't distinguish between branches and commits so some kind of lookup will be necessary.